### PR TITLE
[WIP] [DNM] Adds Silencium as a Bananium counterpart, and makes Reticence mech constructable

### PR DESCRIPTION
**Pull request by Buggy123**
The clown has been getting a lot more features than the mime, but they're supposed to be counterparts. To fix this, I added in a mime-mineral named Silencium, which can only be found on the listening post in space, and made the Reticience mech buildable with it. It can also be used to make invisible, silent airlocks. Probably immensely OP, looking for feedback, do not merge because freeze, etc.

Note that a lot of stuff regarding Silencium and the Reticence is invisible or uses bananium sprites, since I don't have sprites for them and I can't sprite.

TODO:

Add constructable invisible walls [DONE]

Beg someone to make sprites [DONE]

Actually have someone make the sprites

Make a invisible wall creator for the Reticence

Make Silencium-flooring mute any sounds from anything on it(how the fuck would I do this?) OR deafen anyone standing on it (much easier)

Add buildable mime-items to protolathe so Silencium has more uses (any suggestions?)

### CODE REVIEW

**Comment by Anonmare** [comment]
Have we started the fire?

**Comment by PKPenguin321** [comment]
>33 COMMITS
>OVERPOWERED AS FUCK MIME MECH
>OVERPOWERED AS FUCK INVISIBLE AIRLOCKS

👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌👌

**Comment by PKPenguin321** [comment]
>AS I WROTE THE PREVIOUS COMMIT HE MADE A 34TH COMMIT

help i cant breathe

**Comment by xxalpha** [comment]
please squash these commits damn

**Comment by PKPenguin321** [comment]
35 jesus christ help

**Comment by Zilenan91** [comment]
Pls no invisible mechs mr lizard

**Comment by Buggy123** [comment]
I'm squashing the commits as we speak. Do you have any ideas for making this not so fucking OP? I can't really make the walls or airlocks visible without some sprites anyway so I need help there.

**Comment by xxalpha** [comment]
Just make the airlock metal frame invisible but leave the lights as normal.

**Comment by duncathan** [comment]
OH MY GOD

**Comment by Buggy123** [comment]
Are the airlock and airlock lights separate sprites?

**Comment by Anonmare** [comment]
Make it buildable, but you have to be a real mime to use it and maintain your Vow of Silence.

Don't care about the 
>S N O W F L A K E   C O D E

**Comment by Buggy123** [comment]
Commits squashed. Ok, vow of silence check... that'd probably be pretty snowflaky. Plus it would mean that absolutely no one but the mine could use it, which the HONK only requires clown access on a ID.

**Comment by xxalpha** [comment]
>Are the airlock and airlock lights separate sprites?

YESSSSSSSSSSSSSSSSSS

**Comment by Buggy123** [comment]
Alright I'll give it a shot.

**Comment by Buggy123** [comment]
I checked, it already does that actually. The airlock is invisible but the bolted/opening/closing lights are visible, as well as the wire panel if you open it. Do you think I should make it just have the little green/red light panel visible at all times, so it isn't completely invisible when it is opening/closing?

File: code/game/mecha/mecha_construction_paths.dm [hunk 1]
```diff
@@ -863,6 +863,76 @@
...
+					 list("key"=/obj/item/mecha_parts/part/reticence_right_leg),//5
+					 list("key"=/obj/item/mecha_parts/part/reticence_head)
+					)
+
+/datum/construction/mecha/reticence_chassis/action(atom/used_atom,mob/user)
+	return check_all_steps(used_atom,user)
+
+/datum/construction/mecha/reticence_chassis/custom_action(step, atom/used_atom, mob/user)
+	user.visible_message("[user] has connected [used_atom] to the [holder].", "<span class='notice'>You connect [used_atom] to the [holder].</span>")
+	holder.overlays += used_atom.icon_state+"+o"
```
Comment (neersighted) [inline code/game/mecha/mecha_construction_paths.dm]: This is awful and unreadable...

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 2]
```diff
@@ -237,6 +238,32 @@ var/global/list/datum/stack_recipe/clown_recipes = list ( \
 	pixel_x = rand(0,4)-4
 	pixel_y = rand(0,4)-4
 	..()
+	
+/*
+ * Mimeium
+ */
+/obj/item/stack/sheet/mineral/bananium
```
Comment (neersighted) [inline code/game/objects/items/stacks/sheets/mineral.dm]: Copypasta gone wrong. The path is incorrect.

File: code/modules/mining/mint.dm [hunk 3]
```diff
@@ -184,6 +194,14 @@
...
 					src.updateUsrDialog()
 					sleep(5);
+			if("mime")
+				while(amt_mime > 0 && coinsToProduce > 0)
+					create_coins(/obj/item/weapon/coin/mime)
+					amt_mime -= 20
+					coinsToProduce--
+					newCoins++
+					src.updateUsrDialog()
+					sleep(5);
```
Comment (neersighted) [inline code/modules/mining/mint.dm]: bad semicolon

Comment (Buggy123) [inline code/modules/mining/mint.dm]: Wait, how is it bad? That is literally identical to all other ones before/after it, it should be fine.

Comment (neersighted) [inline code/modules/mining/mint.dm]: Just that it should not be here; I would remove it (and the semicolon from all the parents).

Comment (duncathan) [inline code/modules/mining/mint.dm]: We don't use semicolons at the end of lines in our code
Just a style thing

File: code/game/mecha/mecha_construction_paths.dm (hunk 1 above)
Comment (Buggy123) [inline code/game/mecha/mecha_construction_paths.dm]: It's identical to the code for all other mechs...

File: code/modules/mining/mint.dm (hunk 3 above)
Comment (Buggy123) [inline code/modules/mining/mint.dm]: Alright, i'll remove them all.

**Comment by Buggy123** [comment]
Also how is the Reticence OP as fuck? Literally the only thing going for it is it's invisible and it's bullets mute people, so you can kill them silently if they're trapped and unarmed. Otherwise, they do 20 damage and fire slow as fuck, and you're in a slow mech with 100 health.

If you're ambushed by a admittably almost impossible to see mech, you can
A: run away after it shoots you once, because you just took 20 damage and it'll be five seconds until it can fire again.
B: hit it repeatedly with a airtank, you'll probably kill it or damage it's weapon before it can kill you.
C: do literally anything but scream on the comms because it has as much health as a average human and kills slower than a assistant hitting you with a toolbox would.

File: code/game/mecha/mecha_construction_paths.dm (hunk 1 above)
Comment (xxalpha) [inline code/game/mecha/mecha_construction_paths.dm]: ``holder.overlays += "[used_atom.icon_state]+o"``

**Comment by Zilenan91** [comment]
I didnt say it was broken persay, it's certainly beatable, it's just annoying to fight because of the invisibility. I was overreacting a bit there for comedy because it seemed appropriate at the time so I apologize.

File: code/game/mecha/mecha_construction_paths.dm (hunk 1 above)
Comment (Buggy123) [inline code/game/mecha/mecha_construction_paths.dm]: Which is the same as the custom_action for every other mech.

Comment (xxalpha) [inline code/game/mecha/mecha_construction_paths.dm]: Don't stick to bad code, mate.

**Comment by Buggy123** [comment]
Its fine, I just thought it was a serious concern. By the way, would you happen to know why exactly the Reticence is transparent? As far as I can tell the the sprite itself isn't partially transparent, and nowhere in reticence.dm does it mention anything about alpha. It does set the color to #878787, but it doesn't set the alpha.

I can tell ingame it has a alpha of 21, but I don't know where it sets that.

**Comment by Buggy123** [comment]
I updated the walls and airlocks, they now use the normal wall/airlock icon but they're almost completely invisible like the Reticence. This essentially means that silencium walls are like more transparent glass that can't be broken with normal weapons, and airlocks are like more transparent glass airlocks that are also airtight.

File: code/game/mecha/mecha_construction_paths.dm (hunk 1 above)
Comment (drynwyn) [inline code/game/mecha/mecha_construction_paths.dm]: tbh I'd file this under  "the code is fucked up but that's beyond the scope of this PR, which isn't making things worse."

**Comment by Dorsisdwarf** [comment]
:-1: 

stop

adding

admin

items

to

rnd

**Comment by Buggy123** [comment]
Its a fucking shitty admin item tbh.

**Comment by drynwyn** [comment]
reactive tesla armor

**Comment by Buggy123** [comment]
Brb not adding reactive tesla armor to R&D.

**Comment by Shadowlight213** [comment]
:-1: Just stop. seriously. Admin items were not meant to be obtained in game. Rnd already has tons of op shit already.

**Comment by The-Albinobigfoot** [comment]
Why don't we add Deathsquad gear to cargo while we're at it?
:-1:

**Comment by swankcookie** [comment]
:+1: do you need silencium sprites?

**Comment by swankcookie** [comment]
:+1:

**Comment by Buggy123** [comment]
I'll finish this as soon as i figure out how to make walls transparent without leaving a weird black spot on the ground.

**Comment by GunHog** [comment]
Walls are turfs, are they not? I am not sure it is going to be so simple to fix if so.

**Comment by Buggy123** [comment]
Yeah, I may have to just make it a subtype of glass that can't be broken and is deconstructed the same way as a wall.

**Comment by KorPhaeron** [comment]
The simple fix is a wall that has the same icon_state as a plasteel floor.

Not sure if the PR is a good idea though

**Comment by KazeEspada** [comment]
This PR is fine. Just make silencium rare.

**Comment by neersighted** [comment]
The easy way to fix this is to set the wall's icon_state to the icon_state of the turf it replaces

**Comment by KorPhaeron** [comment]
Stale, let me know if you want it re-opened/are going to finish it

**Comment by Buggy123** [comment]
Fair enough.

**Pull request closed without merging**